audiomate/corpus/io/zamia_speech.py

```diff
@@ -0,0 +1,331 @@
+        for file_url in file_urls:
+            file_name = os.path.basename(file_url)
+            target_file_path = os.path.join(target_path, file_name)
+            url_to_target[file_url] = target_file_path
+
+        dl_result = download.download_files(url_to_target, num_threads=1)
+
+        downloaded_files = []
+        for url, status, path_or_msg in dl_result:
+            if status:
+                downloaded_files.append(path_or_msg)
+            else:
+                logger.info("Download failed for url %s", url)
+
+        return downloaded_files
+
+    # ==============================================================================================
+
+    @staticmethod
+    def extract_files(file_paths, target_path):
+        """ Unpack all files to the given path """
+
+        os.makedirs(target_path, exist_ok=True)
+        extracted = []
+
+        for file_path in file_paths:
+            with tarfile.open(file_path, "r") as archive:
+                archive.extractall(target_path)
+
+            file_name = os.path.splitext(os.path.basename(file_path))[0]
+            extracted.append(os.path.join(target_path, file_name))
+
+        return extracted
+
+
+# ==================================================================================================
+
+
+class ZamiaSpeechReader(base.CorpusReader):
+    """ Reader for collections of zamia speech audio data.
+    The reader expects extracted .tgz files in the given folder.
+    Most of the code is take from the Voxforge reader as the structure is similar """
+
+    @classmethod
+    def type(cls):
+        return "zamia-speech"
+
+    def _check_for_missing_files(self, path):
+        return []
+
+    # ==============================================================================================
+
+    def _load(self, path):
+        corpus = audiomate.Corpus(path=path)
+
+        for dir_path in sorted(ZamiaSpeechReader.data_folders(path)):
+            item = os.path.basename(dir_path)
+            etc_folder = os.path.join(dir_path, "etc")
+            wav_folder = os.path.join(dir_path, "wav")
+            readme_path = os.path.join(etc_folder, "README")
+
+            # LOAD ISSUER
+            issuer = ZamiaSpeechReader.parse_speaker_info(readme_path)
+
+            if issuer.idx is None or issuer.idx == "anonymous":
+                issuer.idx = item
+
+            # LOAD TRANSCRIPTIONS
+            prompts, prompts_orig = ZamiaSpeechReader.parse_prompts(etc_folder)
+
+            # LOAD FILES/UTTS
+            for file_name in os.listdir(wav_folder):
+                wav_path = os.path.join(wav_folder, file_name)
+                basename, ext = os.path.splitext(file_name)
+                idx = "{}-{}".format(item, basename)
+
+                is_valid_wav = (
+                    os.path.isfile(wav_path)
+                    and ext == ".wav"
+                    and idx not in self.invalid_utterance_ids
+                )
+                if not is_valid_wav:
```

Comment:
I would create a method that first filters all files and then returns a list with (wav_path, basename, idx).
Reduces size of this method and continue is not needed.